src/platforms.js

```diff
@@ -7,22 +7,51 @@
 import * as unix from "./unix.js";
 import * as win from "./win.js";
 
+/**
+ * @constant {string} cygwin The string identifying the OS type cygwin.
+ */
+const cygwin = "cygwin";
+
+/**
+ * @constant {string} msys The string identifying the OS type msys.
+ */
+const msys = "msys";
+
 /**
  * @constant {string} win32 The string identifying Windows systems.
  */
 const win32 = "win32";
 
+/**
+ * Check if the current platform is Windows.
+ *
+ * @param {Object} args The arguments for this function.
+ * @param {string} args.platform The `os.platform()` value.
+ * @param {Object} args.process The `process` values.
+ * @param {Object} args.process.env The environment variables.
+ * @returns {boolean} `true` iff the current platform is Windows.
+ */
+function isWindow({ platform, process }) {
+  return (
+    platform === win32 ||
+    process.env?.OSTYPE === cygwin ||
+    process.env?.OSTYPE === msys
```

Comment:
Optional chaining (`(.?)`) is a feature added in Node v14 ([[1]](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)), since the current version of Shescape [supports `>=10.13.0`](https://github.com/ericcornelissen/shescape/blob/d1f31ccc612cc6bb883958cdec72b386af2894f6/package.json#L67) it cannot be used here.

Reply:
The environment variable should probably not be considered optional (see [this comment](https://github.com/ericcornelissen/shescape/pull/192#discussion_r821068845)) so optional chaining isn't necessary.